### PR TITLE
Add Nd support to clear_border

### PR DESCRIPTION
**Pull request by jmetz**
## Description
Adds 3d (in principal Nd) support to clear border. 

## References
#2086

### CODE REVIEW

**Comment by ahojnnes** [comment]
Please, update the doc string to reflect the new N-D capability.

**Comment by jmetz** [comment]
@ahojnnes - will do - the only reason I hadn't done this is that I saw that e.g. `regionprops` which accepts `3d` as of a while ago still has similar 2d-version of docstring so wasn't sure if that was left intentionally or not.

**Comment by jmetz** [comment]
Umm is there a preference for the docstring? I am considering either e.g. :
```
labels : (N, M[, L,...]) array of int
        Label or binary image.
```
or 
```
labels : ND array of int
        Label or binary image.
```

**Comment by soupault** [comment]
@jmetz as I'm aware, the preferable versions are `(M[, N[, ..., P]][, C])` for nD multichannel and `(M[, N[, ..., P]])` for nD single-channel.

**Comment by jmetz** [comment]
Thanks @soupault ... changed! I realised the second line of the docstring appeared to 
have been left in by mistake and removed that too (as it's not an optional flag) - let me know if I should undo that.

**Comment by ahojnnes** [comment]
LGTM, thanks!

File: skimage/segmentation/_clear_border.py [hunk 1]
```diff
@@ -5,12 +5,10 @@
...
     """Clear objects connected to the label image border.
 
-    The changes will be applied directly to the input.
-
     Parameters
     ----------
-    labels : (N, M) array of int
-        Label or binary image.
+    labels : (M[, N[, ..., P]]) array of int
+        Label or binary image
```
Comment (soupault) [inline skimage/segmentation/_clear_border.py]: This doesn't seem to be correct. I'd put something like:
```
labels : (M[, N[, ..., P]]) array of int or bool
    Imaging data labels.
```

File: skimage/segmentation/_clear_border.py [hunk 2]
```diff
@@ -21,7 +19,7 @@ def clear_border(labels, buffer_size=0, bgval=0, in_place=False):
 
     Returns
     -------
-    labels : (N, M) array
+    labels : (M[, N[, ..., P]]) array
         Cleared binary image.
```
Comment (soupault) [inline skimage/segmentation/_clear_border.py]: `Cleared binary image.`?

Comment (soupault) [inline skimage/segmentation/_clear_border.py]: Also, I'd change the name of output variable from `labels` to `out`.

File: skimage/segmentation/_clear_border.py (hunk 1 above)
Comment (jmetz) [inline skimage/segmentation/_clear_border.py]: @soupault - I didn't change the phrasing of that one.. can change it to `Imaging data labels` if that's preferred though.

File: skimage/segmentation/_clear_border.py (hunk 2 above)
Comment (jmetz) [inline skimage/segmentation/_clear_border.py]: As above - I simply maintained the previous docstrings in those respects, but if your suggested changes are preferred I can certainly include them in this pull request... or would it make more sense to do that in a different pull request?

Comment (soupault) [inline skimage/segmentation/_clear_border.py]: I'll be very grateful if you could change this. In the current redaction the descriptions are too confusing.
If not, I'll file another issue for this.

Comment (jmetz) [inline skimage/segmentation/_clear_border.py]: Sure no problem

**Comment by soupault** [comment]
@jmetz perfect, thanks! Waiting for the test to pass.

**Comment by jmetz** [comment]
Hey @soupault & @ahojnnes  - sorry to push for this but just so this doesn't end up being in conflict with the the repo... do you think it's good to merge?

**Comment by soupault** [comment]
@jmetz thanks for the contribution!

**Pull request merged**